### PR TITLE
Need to update renewIssueEtc when renewing cookie

### PR DESCRIPTION
**Pull request by HaoK**
cc @Tratcher

### CODE REVIEW

**Comment by HaoK** [comment]
Fixes https://github.com/aspnet/Security/issues/628

I'm not sure if we need to handle sliding expiry as well here...

**Comment by Tratcher** [comment]
I think you need to handle sliding refresh.

https://github.com/aspnet/Security/blob/haok/1-6-issue/src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationHandler.cs#L85-L86
If they access the site every 30 seconds and issuedUtc is updated each time, then timeElapsed will always be 30s and timeRemaining will keep shrinking. The sliding expiration won't be renewed until you're within 30s, and if you miss a request in that window then your cookie suddenly expires.

**Comment by HaoK** [comment]
Yeah I thought as much, this cookie refresh/expires code is pretty complicated as its written :/

**Comment by HaoK** [comment]
I think I will try to refactor the cookie refresh/renew logic into some sane methods so this is actually readable and sane...

**Comment by HaoK** [comment]
Updated:
- Moved logic for refresh into two helpers
- We request refresh if explicitly asked for via context.ShouldRenew, or sliding check
- If we renew, we ALWAYS set expires/issues UTC (is that correct/safe?)

File: src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationHandler.cs [hunk 1]
```diff
@@ -37,6 +37,33 @@ private Task<AuthenticateResult> EnsureCookieTicket()
...
+                if (timeRemaining < timeElapsed)
+                {
+                    RequestRefresh(ticket);
+                }
+            }
+        }
+
+        private void RequestRefresh(AuthenticationTicket ticket)
+        {
+            _shouldRenew = true;
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationHandler.cs]: Any idea why the variables all say renew and not refresh? Are they used for anything else?

**Comment by Tratcher** [comment]
:shipit: and consider updating the naming.

File: src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationHandler.cs (hunk 1 above)
Comment (HaoK) [inline src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationHandler.cs]: Not used anywhere else I don't think, I'll rename to refresh, seems fine

**Comment by HaoK** [comment]
5bcc1bfc260978b0bea9b981056dd0beb31df400

**Pull request closed without merging**